File: ClassStudio.UI/Models/Components/SelectViewModel.cs
/*
 * Copyright (c) 2019 João Pedro Martins Neves (shivayl) - All Rights Reserved.
 *
 * ClassStudio is licensed under the GNU Lesser General Public License (LGPL),
 * version 3, located in the root of this project, under the name "LICENSE.md".
 *
 */

using System.Collections.Generic;

namespace ClassStudio.UI.Models.Components
{
    public class SelectViewModel
    {
        public string Id { get; set; }

        public List<OptionViewModel> Options { get; set; }
    }

    public class OptionViewModel
    {
        public string Label { get; set; }

        public string Value { get; set; }
    }
}

File: ClassStudio.UI/Scripts/src/app/models/select.viewModel.ts
/*
 * Copyright (c) 2019-2020 João Pedro Martins Neves (shivayl) - All Rights Reserved.
 *
 * ClassStudio is licensed under the GPLv3.0 license (GNU General Public License v3.0),
 * located in the root of this project, under the name "LICENSE.md".
 *
 */

export class SelectViewModel {

  options: OptionViewModel[];

  constructor( options: OptionViewModel[] ) {
    this.options = options;
  }

}

export class OptionViewModel {

  label: string;
  value: string;

  constructor( label: string, value: string ) {
    this.label = label;
    this.value = value;
  }

}
